### PR TITLE
adding serverless app template

### PR DESCRIPTION
**Pull request by jeffhollan**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Adding new template to deploy a Logic App that calls a function

### CODE REVIEW

**Comment by jeffhollan** [comment]
@bmoore-msft are you able to merge this? Want to use it as a reference in some docs I'm writing. Thanks

File: 101-logic-app-and-function-app/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  "parameters": {
+    "LogicAppName": {
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 80,
+      "metadata": {
+        "description": "Name of the Logic App."
+      }
+    },
+    "LogicAppLocation": {
```
Comment (bmoore-msft) [inline 101-logic-app-and-function-app/azuredeploy.json]: move this to a variable (the list won't work in other clouds)

File: 101-logic-app-and-function-app/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      }
+    },
+    "branch": {
+      "type": "string",
+      "defaultValue": "master",
+      "metadata": {
+        "description": "The branch of the GitHub repository to use."
+      }
+    },
+    "storageAccountType": {
```
Comment (bmoore-msft) [inline 101-logic-app-and-function-app/azuredeploy.json]: move to variable

File: 101-logic-app-and-function-app/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        "[resourceGroup().location]"
+      ],
+      "metadata": {
+        "description": "Location of the Logic App."
+      }
+    },
+    "functionAppName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the function app to create. Must be unique."
```
Comment (bmoore-msft) [inline 101-logic-app-and-function-app/azuredeploy.json]: globally? in the RG? in the subscription?

File: 101-logic-app-and-function-app/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+      }
+    },
+    "functionName": {
+      "type": "string",
+      "defaultValue":  "Echo",
+      "metadata": {
+        "description": "The name of the function that will be called in the Logic App."
+      }
+    },
+    "repoUrl": {
```
Comment (bmoore-msft) [inline 101-logic-app-and-function-app/azuredeploy.json]: any reason this can't be a var?

File: 101-logic-app-and-function-app/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+      "type": "string",
+      "defaultValue": "https://github.com/AzureBytes/functionshttpecho.git",
+      "metadata": {
+        "description": "The URL for the GitHub repository that contains the project to deploy."
+      }
+    },
+    "branch": {
```
Comment (bmoore-msft) [inline 101-logic-app-and-function-app/azuredeploy.json]: any reason this can't be a var?

File: 101-logic-app-and-function-app/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+      ],
+      "metadata": {
+        "description": "Storage Account type"
+      }
+    }
+  },
+  "variables": {
+    "hostingPlanName": "[parameters('functionAppName')]",
+    "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'azfunctions')]",
+    "storageAccountid": "[concat(resourceGroup().id,'/providers/','Microsoft.Storage/storageAccounts/', variables('storageAccountName'))]"
```
Comment (bmoore-msft) [inline 101-logic-app-and-function-app/azuredeploy.json]: simplify and just use resourceId

**Pull request merged**